### ISSUE TITLE
Rename the application

### ISSUE THREAD

**Issue by BryanDollery** [hkdb/app#14]
Your application name is impossible to google. Try searching for help with 'app', or 'linux app', or anything like that and you'll not get anywhere near finding something, anything, even remotely useful.

I know you thought it was clever, and descriptive and all. It is, it's just that really common words like "the", "phone", and "app" don't make good names for things because they become impossible to find.

I really like your application, I really do. it's great, and I'll use it everywhere from now on, because of the level of standardization it provides. Truly useful. But, really badly named.

**Comment by hkdb** [comment]
I agree that it's a bad name but it's not because I think it's clever. It's cause I am not that creative when it comes to naming. With that said, I am open to suggestions on what to rename app to. 🙏

**Comment by jtbrough** [comment]
@hkdb - I tend to agree with @BryanDollery here. Naming is hard, but this is a terrifically powerful tool that is possibly being limited by its lack of discoverability since searching for `app` is an incredibly overloaded term.

If you are interested in naming suggestions, in addition to those above, I'll humbly submit:

- omni package manager `omni`
- omnipack `op`
- crosspack `xp`

or, if you like acronyms:

- Cross OS Repo Engine `core`
- Extensible Package Manager `xpm`
- Cross Repo Unified Experience `crux`

I do quite like Unified Package Manager `upm` as well.

**Comment by hkdb** [comment]
Unfortunately, there's already a [upm](https://github.com/replit/upm) for a different purpose but looks similar enough at a glance for it to be confusing to new comers. Otherwise, as much as @BryanDollery says it's boring, it's still what feels like my favorite for now yet it feels like it's just not quite good enough.

I like `op` a lot too but the full name itself is ughhh (ie. omnipack or even omnipak) and the searchable term would be the full name so it's always front and center. Not sure if I am OK with that name as the creator of this project.

With that said, I have always thought `opm` as a decent choice but regardless if I ambrace the fact that it can be pronounced like opium and I can even play it off in a humorous way as it's a painkiller for package management, OR... fight it off and say it should not be pronounced like opium declaring specifically to disassociate the project from the negative common pronounciation, I am worried that it may still seem inappropriate for some or even offend someone.

Last but not least, it feels so natural to type `app`. So much so that I will likely alias it or change the binary name after installing it for my personal use case if/when I decide on a better  name to change to 😅

**Comment by hkdb** [comment]
There's apparently this too:

[upm](https://github.com/libalis/upm)

**Comment by BurntVoxel** [comment]
I'm definitely voting for OmniPack. But CRUX is an interesting one, both are cool.

**Comment by grafst** [comment]
For many years i dreamed of a cross-plattform abstraction for this, but never built it.

So i'd like to make a suggestion for a name:
**nstall**

like _install_ but shorter and cooler :)

as far as i can tell there is no important project that uses that name (but this requires further investigation)

I would also gift the domain [nstall.it](https://nstall.it) to you if you want. Better it is used than parked for infinity....

**Comment by hkdb** [comment]
@grafst Thanks for the generous offer. I also like the name nstall. However, I do have my concerns which may or may not seem strong at all.

Here's what I am stuck on. One of the reasons why I am stuck on "app" so much is for practical purposes. Typing "app" is just 3 letters. Also "app install" sounds right. On the other hand "nstall install" would be a little weird to me.

With that said, here's a question I have for you and everyone else on this thread. What do you think of the idea of naming the project one thing but keeping the actual command "app"?

Or.... can we come up with a unique enough name that makes everyone happy from both the marketing/search-ability perspective and the practical perspective of typing as little characters as possible while also maintaining some sort of semantics that makes sense (ie. app install)? 

I think the key here for the latter is that it has to be unique enough. Otherwise, I feel like it defeats the purpose. i.e. I like xpm and op a lot but I question if something like that would be searchable.

**Comment by grafst** [comment]
will think about it...

😄

**Comment by hkdb** [comment]
So, looking through my notes, here are some names I came up with before. I will put it here as options that I am not happy with but closer to what would check all the check boxes:

- appx  - x as in eXtended
- appa - app assistant
- appsync - synchronize packages across multiple machines
- helpme - just kidding... but I thought it was funny in a nerdy way... ie. `helpme install neovim`
- zapp - playing off the word zap

**Comment by grafst** [comment]
yes `nstall install` would be weird. In my fantasies I imagined using just `nstall` and then the identifier.

instead of 
`app install kitty` (17 chars)
use
`nstall kitty` (12 chars)

but maybe thats ugly because of reserved words?


Otherwise i really like zapp! :)
zap as in quick
app as in app
--> zapp

only 4 chars and no similar or big projects:
<img width="860" height="1110" alt="Image" src="https://github.com/user-attachments/assets/139ed830-d9f5-4e36-ad7f-21cb948704e9" />

enjoy your day :)

**Comment by BryanDollery** [comment]
> Otherwise i really like zapp! :) zap as in quick app as in app --> zapp

Zap doesn't mean quick... it means destroy. To zapp something is to kill it - "zap a bug". I read the definition online, it does offer an alternative, but I'd hardly say that "moving around quickly" is common - looks more like some people are using the wrong word for zip.

**Comment by grafst** [comment]
oh, I see, my bad.
Thanks for the clarification @BryanDollery

**Comment by Inobak** [comment]
Dumb suggestion i thought was funny 😄 

OwO => Open Ware Orchestrator
